include/LightGBM/c_api.h

```diff
@@ -127,9 +127,9 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetCreateFromFile(const char* filename,
 LIGHTGBM_C_EXPORT int LGBM_DatasetCreateFromSampledColumn(double** sample_data,
                                                           int** sample_indices,
                                                           int32_t ncol,
-                                                          const int* num_per_col,
-                                                          int32_t num_sample_row,
-                                                          int32_t num_local_row,
+                                                          const int64_t* num_per_col,
+                                                          int64_t num_sample_row,
+                                                          int64_t num_local_row,
```

Comment:
Do you need to support int64_t for sample data and data local to 1 machine?  Not opposing, just making sure you want to do that.  If so, you prob need to change the sample_indices to int64_t as well?

Reply:
> Do you need to support int64_t for sample data and data local to 1 machine? Not opposing, just making sure you want to do that. If so, you prob need to change the sample_indices to int64_t as well?

@svotaw Very happy to receive feedbacks.

Not just using 1 machine. I'm not sure, `int/int32_t` is enough for Worker machines, is it enough to Driver machine?